#134209517 search users and send invites

vonvick opened the pull request:
#### What does this PR do?
This PR searches for users from the database and send an invites to registered and unregistered users.

#### Description of Task to be completed?
Search for users from the database and send invites with the URL for them to join the game. The invites can be sent to both registered and unregistered users.

#### How should this be manually tested?
The result of the test can be seen by running the test and viewing the result of the api routes created.

#### Any background context you want to provide?
No background context

#### What are the relevant pivotal tracker stories?
Users should be able to send invites to at most 11 people other than themselves

#### Screenshots (if appropriate)
![screen shot 2016-12-08 at 12 32 38 pm](https://cloud.githubusercontent.com/assets/21035558/21008576/962e6638-bd42-11e6-85c5-2fbdb120e27c.png)

![screen shot 2016-12-08 at 12 33 04 pm](https://cloud.githubusercontent.com/assets/21035558/21008583/a14cf354-bd42-11e6-9251-b74fb8571f2f.png)


#### Questions:
No questions.

Commit:
fix(eslint): add public/js folder to eslintignore

Commit:
fix(eslint): add commonjs and no-unused var to eslint

Commit:
feature(email): add nodemailer module to send mail

install nodemailer module to send mails and invite users to join game.

Commit:
test(invite): write test for the invite user api

Commit:
test(search): write test for the search users api

Commit:
feature(invite): create invite user function that is used in invite route

Commit:
feature(invite): create invite function for use in the invite route

Commit:
feature(search): create the function that queries the db and return the list of users

Commit:
feature(player): add id to player properties

add id to player prperties and use it to check game creator during game with friends

Commit:
feature(routes): add invite and search routes

add invite and search routes to the routes file and refactor and rearrange the code.

Commit:
feature(game): add search, invite routes and popup

add search and invite routes to the frontend. Also add modal popup to show warning to the user

Commit:
feature(game): change player max to 12 in services

Commit:
feature(game): add directives for invite list template

Commit:
feature(game): create invite and search template

create invite and search template to display list of invited users and search box to find list of users

Commit:
feature(auth): add email from game route to help auth

populate the auth form with email if page was visited from  the game invite link

Commit:
feature(game): add the maximum number of player to start game with

Commit:
feature(popup): create popup modal for game page

create popup modal to warn game creator before start of game

Commit:
fix(invite): fix error with using arrow function on invite directive

Commit:
feature(login): edit auth form to redirect back to game screen if the link is from an invite.

Commit:
fix(error): fix mongoose Promise error while running test

Commit:
feature(invite): add sender's name while sending mail

add sender's name when sending a mail to invite user and also hide nodemailer configuration

Commit:
test(test): rewrite test   to effectively test routes

[Finishes #134209517]

Commit:
fixes(eslint): fix houndci issues on flagged files

Commit:
fixes(invite): use process.env to secure the email for sending invites

Commit:
fixes(heroku):update env variable to app.json file

update extra environment variable to app.json to make app run on heroku

[Finishes #134209517]

xmile1 commented on test/api/invite.js:
```diff
@@ -1,13 +1,41 @@
+const mongoose = require('mongoose');
+
+const User = mongoose.model('User');
+
```
**Nitpick**
A line break might not be necessary here

oagboola commented:
@andela-vnwaiwu your branch has conflicts, please resolve

Commit:
fixes(merge): fix merge conflict for route and directive files

fix merge conflict  on routes and directives file while making a pull request

[Finishes #134209517]

oagboola reviewed (approved):
LGTM
